ldelf/ldelf.mk

```diff
@@ -0,0 +1,47 @@
+include mk/cleanvars.mk
+
+# Set current submodule (used for module specific flags compile result etc)
+sm := ldelf
+sm-$(sm) := y
+sm-$(sm)-is-ld := y
+
+link-out-dir$(sm) := $(out-dir)/$(sm)
+
+cppflags$(sm)	:= $(core-platform-cppflags)
+cflags$(sm)	:= $(core-platform-cflags) -fpie -fvisibility=hidden
```

Comment:
Some numbers:
```
size ../out-os-qemu/ldelf/ldelf.elf
   text    data     bss     dec     hex filename
  17623     264      16   17903    45ef ../out-os-qemu/ldelf/ldelf.elf

Size of ELF file: 323712
Size of dumped data, sizeof(ldelf_data): 20480

Stripped ELF:
size ../out-os-qemu/ldelf/ldelf.elf
   text    data     bss     dec     hex filename
  17623     264      16   17903    45ef ../out-os-qemu/ldelf/ldelf.elf

Size of ELF file: 26324
Size of dumped data, sizeof(ldelf_data): 20480

Without -ffunction-sections -fdata-sections and -gc
size ../out-os-qemu/ldelf/ldelf.elf
   text    data     bss     dec     hex filename
  20322     264      16   20602    507a ../out-os-qemu/ldelf/ldelf.elf


Size of ELF file: 327464
Size of dumped data, sizeof(ldelf_data): 24576
```

As you can see stripping doesn't affect the size of binary embedded in TEE Core so I guess there's no point in doing it. Dropping -ffunction-sections -fdata-sections and -gc results in a slightly larger embedded binary. I guess it's because our libraries are not optimized for this way of linking (object files has more than one function each etc).

As for `.ARM.extab.text.*`, I'll add an output section.

Reply:
That's fine. Thanks for looking at the numbers that actually matter ;-)

Reply:
Why do we get `.ARM.extab.text.*` at all? I see that we have those in `tee.elf` also taking up almost 2kB.

Reply:
Sections `.ARM.extab*` contain unwinding information and are complementary `.ARM.exidx*`.

The exception handling spec [1] states that "*An object producer may generate exception-handling table entries (read-only data) in one ELF section, or one section per function. The name of a section containing an exception table fragment must be .ARM.extab optionally followed by further characters.*"

The `-ffunction-sections` flag is what generates one section per function but as the spec says, it's OK to group them in the same output section (which we do for `tee.elf`, and as you suggested, we should do also for `ldelf.elf`).

Not that these tables are not generated when `CFG_UNWIND=n`, which is expected.

[1] http://infocenter.arm.com/help/topic/com.arm.doc.ihi0038b/IHI0038B_ehabi.pdf

Reply:
It seems that "exception-handling table entries" isn't needed in normal C code as it doesn't support exceptions any way. I assume that exception means the C++ type of exception. We should perhaps in a future optimization discard all the unneeded `.ARM.extab*` sections to save some memory.

Reply:
Agreed.